sdks/unity/AgonesSdkClient.cs

```diff
@@ -0,0 +1,179 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Newtonsoft.Json;
+using System.Collections;
+using System.Text;
+using UnityEngine;
+using UnityEngine.Networking;
+
+namespace Agones.SDK
+{
+    /// <summary>
+    /// JSON payload object
+    /// </summary>
+    internal class KeyValueMessage
+    {
+        public string key;
+        public string value;
+    }
+
+    /// <summary>
+    /// Agones SDK Unity Client
+    /// </summary>
+    public class AgonesSdkClient : MonoBehaviour
+    {
+        private const string emptyPayload = "{}";
+        private static byte[] emptyPayloadBytes;
+        private WaitForSeconds wait = new WaitForSeconds(0.5f);
+        private float updateInterval = 0.5f;

```

Comment:
Three fun(?) questions about this bit here:
1. Should `updateInterval` be public, so that people can adjust this via the Component editor?
1. Since we have the capability to grab the `GameServer` configuration from the SDK, should we do that, and then use that to start this Health ping? (then the `updateInterval` would not need to be public)
1. If we are going to set an opinionated health loop - shouldn't we also provide a callback mechanism so that users can provide their own base logic for health pinging? (and maybe also a way to disable the opinionated health ping?). Right now, there is no other choice on health pinging other than what is provided.

Reply:
Answering out-of-order

2: I would say yes, but note that Unity Developers really like being able to modify things directly in the Editor. (Estimating based on personal observation only) it seems like upwards of 1/3 of all the Asset Packages are just custom Editors that provide interfaces for things that engineers can do with code.

If we expect anyone using this package to already be familiar with Agones, Kubernetes, etc, then we can just grab the GameServer configuration.

However, if we want this to be accessible to newbie coders or even designers at some point, then I would suggest a long-term goal of providing some kind of Editor Plugin. This is an area I've spent a lot of time on, so I'd be able to help with that.

1: As above, yes (public) if we want it to be accessible to people less familiar with the console. Another option instead of making this public would just be to add the `[SerializeField]` tag to the `UpdateInterval` property below, that way it shows up in Editor and modifying it calls the `set` method accordingly.

 3: This could be as simple as making the "Health" method virtual, and allowing the user to subclass AgonesSdkClient if they want different Health checking. As of now, the option to disable Health Checking appears to be `LaunchOptions.AgonesEnabled`, because I don't see that being used anywhere else, but if there are circumstances where you want Agones enabled but not Health checking, then yeah it could be a different flag.